dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -36,6 +36,42 @@
 __all__ = ["dpnp_dot", "dpnp_matmul"]
 
 
+def _create_result_array(x1, x2, out, shape, dtype, usm_type, sycl_queue):
+    """
+    Create the result array.
+
+    If `out` is not ``None`` and its features match the specified `shape`, `dtype,
+    `usm_type`, and `sycl_queue` and it is C-contiguous or F-contiguous and
+    does not have any memory overlap with `x1` and `x2`, `out` itself is returned.
+    If these conditions are not statisfied, an empty array is returned with the
+    specified `shape`, `dtype, `usm_type`, and `sycl_queue`.
+    """
+
+    if out is not None:
+        dpnp.check_supported_arrays_type(out)
```

Comment:
I guess the line is not needed, because the same check is done inside `out_usm = dpnp.get_usm_ndarray(out)`, isn't it?
```suggestion
```